pkg/clients/database/configuration/mysql.go

```diff
@@ -0,0 +1,118 @@
+/*
+Copyright 2021 The Crossplane Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package configuration
+
+import (
+	"context"
+	"net/http"
+
+	"github.com/Azure/azure-sdk-for-go/services/mysql/mgmt/2017-12-01/mysql"
+	"github.com/Azure/go-autorest/autorest"
+
+	azuredbv1beta1 "github.com/crossplane/provider-azure/apis/database/v1beta1"
+	"github.com/crossplane/provider-azure/apis/v1alpha3"
+	azure "github.com/crossplane/provider-azure/pkg/clients"
+)
+
+// NOTE: postgresql and mysql structs and functions live in their respective
+// packages even though they are exactly the same. However, Crossplane does not
+// make that assumption and use the respective package for each type, although,
+// they both share the same SQLServerParameters and SQLServerObservation objects.
+// https://github.com/Azure/azure-sdk-for-go/blob/master/services/mysql/mgmt/2017-12-01/mysql/models.go
+// https://github.com/Azure/azure-sdk-for-go/blob/master/services/postgresql/mgmt/2017-12-01/postgresql/models.go
+
+// MySQLConfigurationAPI represents the API interface for a MySQL Server Configuration client
+type MySQLConfigurationAPI interface {
+	Get(ctx context.Context, s *azuredbv1beta1.MySQLServerConfiguration) (mysql.Configuration, error)
+	CreateOrUpdate(ctx context.Context, s *azuredbv1beta1.MySQLServerConfiguration) error
+	Delete(ctx context.Context, s *azuredbv1beta1.MySQLServerConfiguration) error
+	GetRESTClient() autorest.Sender
+}
+
+// MySQLConfigurationClient is the concreate implementation of the MySQLConfigurationAPI interface for MySQL that calls Azure API.
+type MySQLConfigurationClient struct {
+	mysql.ConfigurationsClient
+}
+
+// NewMySQLConfigurationClient creates and initializes a MySQLConfigurationClient instance.
+func NewMySQLConfigurationClient(cl mysql.ConfigurationsClient) *MySQLConfigurationClient {
+	return &MySQLConfigurationClient{
+		ConfigurationsClient: cl,
+	}
+}
+
+// GetRESTClient returns the underlying REST client that the client object uses.
+func (c *MySQLConfigurationClient) GetRESTClient() autorest.Sender {
+	return c.ConfigurationsClient.Client
+}
+
+// Get retrieves the requested MySQL Configuration
+func (c *MySQLConfigurationClient) Get(ctx context.Context, cr *azuredbv1beta1.MySQLServerConfiguration) (mysql.Configuration, error) {
+	return c.ConfigurationsClient.Get(ctx, cr.Spec.ForProvider.ResourceGroupName, cr.Spec.ForProvider.ServerName, cr.Spec.ForProvider.Name)
```

Comment:
Looks like we can get rid of the `spec.forProvider.Name` field, however, it will yield a suboptimal UX: We can use the last path component of a non-empty `external-name` or if `external-name` is empty, use `metadata.name` as the configuration name (in place of `spec.forProvider.Name`). However, the constraints posed on `metadata.name` may prevent specifying all configuration names, and because the `external-name` incorporates the Azure subscription ID as a path component, the `external-name` needs to be in the format: `/subscriptions/<subscription ID>/resourceGroups/<resource group>/providers/Microsoft.DBforMySQL/servers/<server name>/configurations/<config name>`. And in such scenarios, I think it's not a good UX to have to specify this as an `external-name`. Thus, giving the name via `spec.forProvider.Name` provides a better UX IMO. This is also the case in PostgreSQL server configuration MR.

Reply:
@ulucinar thank you for this comment